apps/openmw/mwgui/journalviewmodel.cpp

```diff
@@ -305,18 +305,46 @@ struct JournalViewModelImpl : JournalViewModel
         visitor (toUtf8Span (topic.getName()));
     }
 
-    void visitTopicNamesStartingWith (char character, std::function < void (const std::string&) > visitor) const
+    void visitTopicNamesStartingWith (int index, std::function < void (const std::string&) > visitor) const
     {
         MWBase::Journal * journal = MWBase::Environment::get().getJournal();
 
         for (MWBase::Journal::TTopicIter i = journal->topicBegin (); i != journal->topicEnd (); ++i)
         {
-            if (i->first [0] != Misc::StringUtils::toLower(character))
-                continue;
+            unsigned char byte1 = i->first[0];
+            // First, check for two-byte UTF-8 symbols, e.g. Cyrillic ones
+            // TODO: check which language journal index is using
+            if ((byte1 == 0xd0 || byte1 == 0xd1) && i->first.length() >= 2)
```

Comment:
Wouldn't it be easier to convert the topic name to utf32 instead of messing with individual bytes?

Reply:
https://github.com/OpenMW/openmw/blob/11c4aed4e568f0b88d5a0d746351fc90c6ddd15e/components/misc/utf8stream.hpp

https://github.com/OpenMW/openmw/blob/7a389f7ab117acad15d813a33f63877f45da98ef/components/fontloader/fontloader.cpp#L25

Reply:
Do you mean the MyGUI::UString::utf32string?